airbyte-cdk/bulk/core/extract/src/main/kotlin/io/airbyte/cdk/read/RootReader.kt

```diff
@@ -58,60 +57,44 @@ class RootReader(
     val streamStatusManager = StreamStatusManager(stateManager.feeds, outputConsumer::accept)
 
     /** Reads records from all [Feed]s. */
-    suspend fun read(listener: suspend (Map<Feed, Job>) -> Unit = {}) {
+    suspend fun read(listener: suspend (Collection<Job>) -> Unit = {}) {
+        readFeeds<Global>(listener)
+        readFeeds<Stream>(listener)
+    }
+
+    private suspend inline fun <reified T : Feed> readFeeds(
+        crossinline listener: suspend (Collection<Job>) -> Unit,
+    ) {
+        val feeds: List<T> = stateManager.feeds.filterIsInstance<T>()
+        log.info { "Reading feeds of type ${T::class}." }
+        val exceptions = ConcurrentHashMap<T, Throwable>()
         supervisorScope {
```

Comment:
This is going to run all Global feeds before all Stream feeds? am I reading this correctly?

Reply:
yes